src/github/client/github-client.ts

```diff
@@ -26,5 +29,22 @@ export class GitHubClient {
 			this.restApiUrl = `${baseUrl}/api/v3`;
 			this.graphqlUrl = `${baseUrl}/api/graphql`;
 		}
+
+		const proxyConfig = this.getProxyConfig(this.restApiUrl);
+
+		this.axios = axios.create({
+			baseURL: this.restApiUrl,
+			transitional: {
+				clarifyTimeoutError: true
+			},
+			... proxyConfig
+		});
 	}
+
+	private getProxyConfig = (baseUrl: string): Partial<AxiosRequestConfig> => {
+		if (baseUrl.includes("github.internal.atlassian.com")) {
+			return NO_PROXY_CONFIG;
+		}
+		return OUTBOUND_PROXY_CONFIG;
+	};
```

Comment:
Moved the axios client to the `GitHubClient` parent class and added the proxy config (or not, depending on the `baseUrl`).